doc/api/test_runner.md

```diff
@@ -0,0 +1,275 @@
+# Test runner
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/test_runner.js -->
+
+The `test_runner` module facilitates the creation of JavaScript tests that
+report results in [TAP][] format. To access it:
+
+```js
+const test = require('test_runner');
+```
+
+Tests created via the `test_runner` module consist of a single function that
+executes either synchronously or asynchronously. Synchronous tests are
+considered passing if they do not throw an exception. Asynchronous tests return
+a `Promise`, and are considered passing if the returned `Promise` does not
+reject. The following example illustrates how tests are written using the
+`test_runner` module.
+
+```js
+const assert = require('assert');
+const test = require('test_runner');
+
+test('synchronous passing test', (t) => {
+  // This test passes because it does not throw an exception.
+  assert.strictEqual(1, 1);
```

Comment:
As far as I can understand the assert operations do not cause output. The output of passing assertions however has proven to be **very** helpful to me trying to figure how/why things fails. Naturally the assert operations should only cause output within the test. Maybe a `t.strictEqual` API variant that logs the statements may be possible?

Reply:
I don't plan to provide assertion library integration. Anyone else is welcome to add this feature in the future though - I think it would be pretty simple to do.

Reply:
> I think it would be pretty simple to do.

😅 I wouldn't know where to start.

Reply:
The first thing that comes to mind for me (there may be more optimal solutions), assuming you're trying to add `t.strictEqual()`:

In `TestContext` add a `strictEqual()` function that calls `this.#test.createSubtest()`. Instead of taking a user provided function for the subtest, use a function that wraps `assert.strictEqual()`.